astropy/coordinates/tests/test_representation.py

```diff
@@ -997,19 +1116,34 @@ def test_getitem_scalar(self):
 
     def test_transform(self):
 
-        s1 = CartesianRepresentation(x=[1, 2] * u.kpc, y=[3, 4] * u.kpc, z=[5, 6] * u.kpc)
+        ds1 = CartesianDifferential(d_x=[1, 2] * u.km / u.s,
+                                    d_y=[3, 4] * u.km / u.s,
+                                    d_z=[5, 6] * u.km / u.s)
+        s1 = CartesianRepresentation(x=[1, 2] * u.kpc, y=[3, 4] * u.kpc,
+                                     z=[5, 6] * u.kpc, differentials=ds1)
 
         matrix = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
 
+        # transform representation & get comparison (thru CartesianRep)
         s2 = s1.transform(matrix)
+        ds2 =  s2.differentials["s"]
+
+        dexpected = CartesianDifferential.from_cartesian(
```

Comment:
You're not using `dexpected` (I think both comparing it *as well* and just omitting it are fine)

Reply:
I added comparisons.